executor/runtime/docker/docker.go

```diff
@@ -1427,6 +1436,98 @@ func (r *DockerRuntime) statusMonitor(cancel context.CancelFunc, c runtimeTypes.
 	}
 }
 
+// startOtherUserContainers launches the other user containers, only looking
+// any container objects in the pod *after* the first one, converting the
+// v1.Container spec into something docker can understand, and then
+// running that container.
+func (r *DockerRuntime) startOtherUserContainers(ctx context.Context, pod *v1.Pod, mainContainerID string) error {
+	if pod == nil || pod.Spec.Containers == nil {
+		return nil
+	}
+	otherUserContainers := pod.Spec.Containers[1:]
```

Comment:
I guess this is temporary until we pass in more information about which containers are "user" vs "sidecar", right?  It's ok for now, but as soon as we actually support this in the API, having the first container be the main one won't be something we can rely on.

Reply:
Yep. When your pod stuff hits in the API, we can look at the schema and not assume the first container is the main one.
(or something like that).

But all the other logic about pulling, creating will be the same. (should be)

*Then* lastly, when it comes to startup+healthchecks, that is where we'll care about which are user workloads vs sidecars, and we can be more precise about startup order.

Reply:
It also occurs to me that you'll need to make sure that the pod-backed `Container` implementation (in `executor/runtime/types/pod.go` also picks the right user container. That code currently uses [GetUserContainer() from titus-kube-common](https://github.com/Netflix/titus-kube-common/blob/c6a533ee35ad49e3157579fe73197f4cc1639030/pod/containers.go#L7).

Reply:
Wait, I thought you already did that? I rebased on master on your changes and I think that is in.